src/lib/Generator.class.ts

```diff
@@ -81,14 +67,14 @@ class Generator extends EventEmitter {
     if (this.$thisList.length >= (batchSize ?? this.batchSize)) {
       if (this.source === '') this.source = getEngineSource(this.endpoint)
       const unionQuery = getSPARQLQuery(getSPARQLQueryString(this.query), "construct");
-      const union: UnionPattern = { type: 'union', patterns: [] }
+      const patterns = unionQuery.where ?? [];
+      const valuePatterns: ValuePatternRow[] = []
       for (const $this of this.$thisList) {
         this.iterationsProcessed++
-        const group: GroupPattern = { type: 'group', patterns: [...unionQuery.where ?? []] }
-        group.patterns.unshift(this.$thisToBind($this))
-        union.patterns.push(group)
+        valuePatterns.push({'?this': $this})
       }
```

Comment:
Isn't this more readable / concise with a `map`? If you think writing out the for-loop is more readable, you can resolve this comment without making any changes.

Reply:
Oh I only now read your slack message. You can also count inside of the map, but just do whatever you think is most understandable.